### ISSUE TITLE
about using a reference GAN model

### ISSUE THREAD

**Issue by cxw-droid** [DingfanChen/GAN-Leaks#2]
Hi,

Thank you for your paper and for sharing the code.

In the paper, a reference GAN model is used for a calibrated attack due to the representation error. I have a question here. It is understandable some query sample may have a bigger distance `L(x, R(x|G))` because of its complicated representation, then its representation from a reference GAN model usually will have a bigger distance too. How can the difference of two bigger distances (i.e. the distance after calibration) will be a better metric? My understanding is the difference of two bigger distances will still be a bigger one in general. Would you like to elaborate on your reasoning of the calibrated distance? Thanks!

**Comment by DingfanChen** [comment]
Hi, thanks for your interest.
Indeed, the high-level concept is that certain data samples, particularly those that exhibit complex representations or are underrepresented, will yield a high reconstruction error when processed by a target model. Without proper calibration, these member data samples can not be detected due to their comparatively large reconstruction errors.

By calibration, the reconstruction error of such data samples on a reference model will also be large. Consequently, when we calculate the difference between the reconstruction errors from the target and the reference models, we tend to obtain a smaller value—subtracting a larger number results in a smaller value. This smaller calibrated reconstruction error could potentially make the underrepresented or complex samples detectable by an attacker.

While this method might not entirely eliminate the original false positives (non-members detected as members due to a small reconstruction distance by the target model) and false negatives (members classified as non-members due to a large reconstruction distance by the target model), it can correct many such misclassifications. Specifically, it allows for the possibility that a false positive (small - small) might result in a larger value than a false negative (large - large), which could never occur without calibration.

Hope these help clarify the point. The visual illustration of Figure 3 in the [paper](https://arxiv.org/pdf/1909.03935.pdf) may also be helpful for understanding. Please just share further questions if anything remains unclear.

**Comment by cxw-droid** [comment]
Thank you very much for your reply.

My understanding is: in general, it is not known whether the difference of two bigger is bigger or smaller (e.g., 3 - 2 > 0.3 - 0.2, but 2 - 3 < 0.3 - 0.2,  assuming 3, 2 are the bigger representations of a more complex data sample and 0.3, 0.2 are the smaller representations of a simple sample).

I guess possibly it is because the target model is better trained than the reference model? If so, then in the target model, the bigger value for the complex sample may get a little smaller and the smaller value for the simple sample may get a little bigger because the target model is better trained and have improved/more accurate representations. So, e.g., in the paper, for white-box setting, are the PGGAN reference model and target model trained with the same configurations?

**Comment by DingfanChen** [comment]
- Yes, in general, it is not clear whether the difference between two bigger values is bigger or smaller.
- The main point is: without calibration, the complex samples will always be ```big``` (even for the member samples) and the simple samples will always be ```small```. While with calibration, it's very likely that the complex member samples will no longer have ```big``` calibrated difference and the simple nonmember samples will no longer have ```small``` calibrated difference, which will reduce the FPR and TPR.
- It might not be accurate to state "target model is better trained than the reference model" as they are trained under the same configuration and should be "equivalently well" for representing their own training distributions (and the training datasets for the target and reference model are *disjoint* to minimize potential confounding factors). Moreover, the role of calibration does *not* depend on the different degrees of training between the target and reference model. The only requirement is that the reference model should be sufficiently well-trained to faithfully reflect the sample difficulty.
